cypress/test/components/CaseStudyCard.cy.js

```diff
@@ -20,4 +23,29 @@ describe('CaseStudyCard Component', () => {
 
     cy.get('.rounded-md').should('not.exist');
   });
+
+  //tests for the Adopters section
+  it('renders the Adopters section with adopters data', () => {
+    mount(<Casestudies />);
+    
+    cy.get('[data-testid="Adopters"]').should('exist');
+
+    AdoptersList.forEach((entry, index) => {
+      cy.get(`table tbody tr:eq(${index}) td:eq(0)`).should('have.text', entry.companyName);
+      cy.get(`table tbody tr:eq(${index}) td:eq(1)`).should('have.text', entry.useCase);
+
+      entry.resources.forEach((resource, resourceIndex) => {
+        cy.get(`table tbody tr:eq(${index}) td:eq(2) ul li:eq(${resourceIndex}) a`)
+          .should('have.attr', 'href', resource.link)
+          .should('have.text', resource.title);
+      });
+    });
+  });
+
+  it('does not render anything when adopters array is empty', () => {
+    // Modify Casestudies component to pass an empty AdoptersList
+    mount(<Casestudies adopters={[]} />);
```

Comment:
you haven't add any prop of adopters in `CaseStudies` component, how are you passing it to check that adopters should be empty?

Reply:
> you haven't add any prop of adopters in `CaseStudies` component, how are you passing it to check that adopters should be empty?

@akshatnema i was testing few things forgot to remove it from the code 😅.  i have updated the tests but it still failing, i don't why it is failing